### ISSUE TITLE
Fix EVerest Testing Environment 

### ISSUE THREAD

**Issue by jhoshiko** [EVerest/everest-demo#30]
When trying to run the automated testing demo, the tests fail to run because pytest is not installed:
```
mqtt-server-1  | 1710442039: mosquitto version 2.0.10 starting
mqtt-server-1  | 1710442039: Config loaded from /mosquitto/config/mosquitto.conf.
mqtt-server-1  | 1710442039: Opening ipv4 listen socket on port 1883.
mqtt-server-1  | 1710442039: Opening websockets listen socket on port 9001.
mqtt-server-1  | 1710442039: mosquitto version 2.0.10 running
manager-1      | ./run-test.sh: line 3: pytest: not found
manager-1 exited with code 127
```
However, after installing pytest (specifically `pytest-asyncio`), the tests still fail because the `everest.testing` module is missing:
```
__________________________________________________________________ ERROR collecting core_tests/startup_tests.py ___________________________________________________________________ImportError while importing test module '/ext/source/tests/core_tests/startup_tests.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
core_tests/startup_tests.py:12: in <module>
    from everest.testing.core_utils.everest_core import EverestCore, Requirement
E   ModuleNotFoundError: No module named 'everest.testing'
============================================================================= short test summary info =============================================================================ERROR core_tests/startup_tests.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!================================================================================ 1 error in 0.14s =================================================================================
```
I can see that `everestpy` and `edm_tool` are both installed, and I can import everest.

**Comment by jhoshiko** [comment]
Using the `pkgutil` package, I can confirm that `everest.framework` exists, but I still do not see `everest.testing`

**Comment by jhoshiko** [comment]
@couryrr-afs Shankari recommended reaching out to you for assistance. Do you know what might be happening?

**Comment by jhoshiko** [comment]
Looks like others have had issues with the `everestpy` package missing but nothing mentioning `everest.testing` missing. They were able to fix it by rebuilding with an updated `bullseye-toolchain.tgz`. Not sure what we used, but this might be why `everest.testing` is missing.

**Comment by couryrr-afs** [comment]
@jhoshiko that looks to be a missing python dependency. I am attempting to rebuild the docker image calling `pip install` to verify if that is the missing step.

**Comment by couryrr-afs** [comment]
Update on the status of the issue. There was a missing step for installing pytest which solves the initial problem outline here. There is a branch on the US-JOET fork that provides the needed updates couryrr/updating-dockerfile-for-testing. However, as we attempt to run the test there appears to have been a breaking change in a new release of everest. Currently, attempting to track down the breaking change.

**Comment by shankari** [comment]
@couryrr-afs the UI-based tests are running (or will be running after I make push some changes). You might want to try replicating them in the automated tests.

**Comment by couryrr-afs** [comment]
Just wanted to give a quick status update. To get by the pytest issue described here it was simply adding the package along with pytest-asyncio via pip. After making this change there were issues with the packages that were being pulled in via the build-kit image and what we had been building. I moved the version up for the everest tag which appears to have solved all of the pytest issue from what I can tell. Unfortunately, when testing the full process with the bash script I am running into a new issue below. I am working to track down the cause of this.

```
DEBUG    root:everest_core.py:208   Syscall pipe2() failed (Invalid argument), exiting
manager-1      | DEBUG    root:everest_core.py:208   
manager-1      | WARNING  root:everest_core.py:215 EVerest stopped with return code: 1
manager-1      | DEBUG    root:everest_core.py:217 EVerest output stopped
```

**Comment by shankari** [comment]
@couryrr-afs have you tried any of the existing demos as a template? I am able to run most of the existing demos and certainly the scenarios being tested in these tests through the UI. The only tests that are failing are ISO 15118-2 + OCPP 201.

>  After making this change there were issues with the packages that were being pulled in via the build-kit image and what we had been building.

It would be helpful to understand what these changes were and how they are different from the script currently in the Dockerfile.

I have (today morning) updated the demo images to the RC1 tag. I would suggest sticking to that tag for easier reproducibility and coordination within the community

**Comment by couryrr-afs** [comment]
Currently we are able to install pytest so the initial reported issue was a fairly simple fix. However, we cannot successfully get the test to run at this point in the docker container. I was able to replicate the setup in the devcontainer here: https://github.com/EVerest/everest-utils/tree/main/docker/everest-playground. Inside this devcontainer we are able to successfully run the test on the 2024.3.0-rc2 tag. 

When moving this back over to the docker container for the test we receive a failure as outline above. There is a zulip thread here around getting more details from the manager https://lfenergy.zulipchat.com/#narrow/stream/417679-EVerest.3A-Testing-.26-CI/topic/Verbose.20Manager.20Error.20Output/near/430684671. At the moment this has been unsuccessful.

At this point we are considering this blocked by this issue: https://github.com/EVerest/everest-core/issues/619

**Comment by couryrr-afs** [comment]
Per a comment on the issue https://github.com/EVerest/everest-framework/issues/180 I have attempted to rebuild and run the container in a proper linux environment. Currently, the build process is using a high amount of resource that my system cannot sustain. I am attempting to find an alternative to verify the solution.

**Comment by couryrr-afs** [comment]
The demo was successfully ran in docker on a PopOS host per the details here: https://github.com/EVerest/everest-framework/issues/180#issuecomment-2066536204

It appear we were communicating in multiple github issues. I will keep this one up-to-date going forward.

**Comment by couryrr-afs** [comment]
@shankari 

Moving conversation for the demo work here. This is in response to https://github.com/EVerest/everest-framework/issues/180#issuecomment-2077869878

> @couryrr-afs I do not plan to run the demo to test this out until I know the exact scenario in which it passes. I don't want to spend my time working through various configurations.
> 
> Can you please list out the known working configuration (host OS/VM OS/docker OS...) so that I can start there and try to go from working to working?

For the issue opened here tests could not be ran because pytest was not installed. This issue was resolved with the updated `install.sh`

The manager however was booting but then dying with a cryptic error:

> [ERRO] manager         int main(int, char**) :: Main manager process exits because of caught exception: DEBUG    root:everest_core.py:208   Syscall pipe2() failed (Invalid argument), exiting

A successful build of the image and running container using Docker CE was completed on:

- Host OS: PopOS which is based on Ubuntu 22.04
- Chip was Intel i7-9750H CPU @ 2.60GHz
- Docker image:
    - Built locally on US-JOET/everest-demo using branch `couryrr/updating-dockerfile-for-testing` (commit hash: c7106eff70bbf30a0f8196b913e68190065c0d98)
    - The manager was built using manager/Dockerfile with the manager directory as the build directory
    - Image built on ghcr.io/everest/build-kit-alpine:latest with image digest 7494bd6624aee3f882b4f1edbc589879e1d6d0ccc2c58f3f5c87ac1838ccd1de
